src/core/project/qgsprojectviewsettings.h

```diff
@@ -104,6 +104,20 @@ class CORE_EXPORT QgsProjectViewSettings : public QObject
      */
     void setPresetFullExtent( const QgsReferencedRectangle &extent );
 
+    /**
+     * Sets wheter the project's preset full extent should be loaded when the project is loaded.
+     *
+     * \since QGIS 3.44
+     */
+    void setMainCanvasOpensAtProjectExtent( bool state );
```

Comment:
I'd suggest a different wording here, something like setRestoreFullExtentOnProjectLoad or something. I'm not sure I have the right phrasing but I think we can find something better here :)